File: src/main/java/szewek/flux/util/gift/GiftData.java
package szewek.flux.util.gift;

import net.minecraft.item.ItemStack;

import java.util.List;
import java.util.stream.Collectors;

public final class GiftData {
	public final int boxColor, ribbonColor;
	private final List<ItemStack> stacks;

	public GiftData(int boxColor, int ribbonColor, List<ItemStack> stacks) {
		this.boxColor = boxColor;
		this.ribbonColor = ribbonColor;
		this.stacks = stacks;
	}

	public List<ItemStack> getStacks() {
		return stacks.stream().map(ItemStack::copy).collect(Collectors.toList());
	}

}

File: src/main/java/szewek/flux/util/gift/GiftData.kt
package szewek.flux.util.gift

import net.minecraft.item.ItemStack
import net.minecraft.util.text.ITextComponent
import net.minecraft.util.text.TranslationTextComponent
import java.util.function.Supplier
import javax.annotation.Nonnull

class GiftData(
        @JvmField val name: String,
        @JvmField val day: Int,
        @JvmField val month: Int,
        @JvmField val boxColor: Int,
        @JvmField val ribbonColor: Int,
        private val stacks: List<ItemStack>
) : IGift {
    override fun getStacks() = stacks.map(ItemStack::copy)

    override fun getText() = TranslationTextComponent("flux.gift.text.$name")

}